Bottom: 7b66918a7dc384692d5431aed2b73634df79ed6f
Top:    9ad0aa95aa46c022d13d8aebf17f0c0220743867
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-16 00:48:06 +0000

Refresh of crosshair-c-be-paranoid-about-

---

diff --git a/src/crosshair.c b/src/crosshair.c
index 3c6c46e..a644cc7 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -1026,8 +1026,8 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
   if (ans != NO_TYPE)
     {
       PadType *pad = (PadType *)ptr2;
-      check_snap_object (&snap_data, (pad->Point1.X + pad->Point2.X) / 2,
-                                     (pad->Point1.Y + pad->Point2.Y) / 2,
+      check_snap_object (&snap_data, pad->Point1.X + (pad->Point2.X - pad->Point1.X) / 2,
+                                     pad->Point1.Y + (pad->Point2.Y - pad->Point1.Y) / 2,
                          true);
     }
